wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -1,178 +1,817 @@
-- Knowledge in [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet), specifically [Github-Flavoured Markdown(GFM)](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/ "GitHub Help on Writing on GitHub").
+- A [GitHub account](https://help.github.com/articles/signing-up-for-a-new-github-account/ "GitHub Help in Signing up an account").
+- Knowledge in [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet "adam-p's cheatsheet for Markdown"), specifically [GitHub-Flavoured Markdown(GFM)](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/ "GitHub Help on Writing on GitHub").
+- [Understanding the Git's workflow (feature-branch workflow)](https://help.github.com/articles/what-is-a-good-git-workflow/ "GitHub Help on Git workflow").
+  - [A simplified visual explanation provided by GitHub Guides](https://guides.github.com/introduction/flow/ "Interactive visual Git workflow").
+- Able to write in English, preferably British English for consistency.
+  - Everything not related to the page content itself (commit messages, pull requests, filenames, etc) **MUST** be written in English.
 
 Optionally, if you plan to make changes locally:
 
-- A text editor with Markdown syntax highlighting.
-  - We either use Microsoft's [Visual Studio Code](https://code.visualstudio.com) or Github's [Atom](https://atom.io).
-  - Any text editor will suffice if you do not need the syntax highlighting.
-- A Markdown previewer with **GFM** support. 
-  - We use [Shiba](https://github.com/rhysd/Shiba), or Visual Studio Code/Atom's built-in GFM previewers.
-- Basic knowledge in VCS (Version control system), particularly [Git](https://git-scm.com/ "Git").
+- A text editor.
+  - Markdown syntax highlighting is an optional preference.
+    - We either use Microsoft's [Visual Studio Code](https://code.visualstudio.com "Visual Studio Code"), Adobe's [Brackets](http://brackets.io/ "Brackets"), or GitHub's [Atom](https://atom.io "Atom").
+- A Markdown previewer with **GFM** support.
+  - We use [Shiba](https://github.com/rhysd/Shiba "Shiba") or Visual Studio Code's/Atom's built-in GFM previewers.
+- Basic knowledge in VCS (Version control system), particularly in using [Git](https://git-scm.com/ "Git").
+
+## Forking osu-wiki repository
 
-## Forking
- 
- To be able to make changes to the repository, you must first create your own copy of the repository. This is known as "forking".
- 
- Go to the [osu!wiki repository](https://github.com/ppy/osu-wiki).
- 
- You should see something like this.
+To make changes to the osu-wiki repository, you must first create your own copy of the repository.
+This is known as "forking".
 
-![osu-wiki Repository Snapshot](3.jpg "Snapshot of the osu!Wiki Repository")
+Go to the [osu!wiki repository](https://github.com/ppy/osu-wiki "osu-wiki in GitHub").
```

Comment:
nitpicking, whenever I refer to osu-wiki (the repo), I use `osu-wiki repo`, not `osu!wiki repo`